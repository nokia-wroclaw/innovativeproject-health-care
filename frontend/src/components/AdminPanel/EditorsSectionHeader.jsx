import React, { useState } from "react";
import { Header, Card, Grid } from "semantic-ui-react";
import SectionIcon from "../common/SectionIcon/";
import AddEditorForm from "./AddEditorForm";

export const EditorsSectionHeader = () => {
  const [showInput, setShowInput] = useState(false);
  return (
    <React.Fragment>
      <Card.Content>
        <Grid>
          <Grid.Row columns={3}>
            <Grid.Column width={13} floated="left">
              <Header as="h4">Editors</Header>
            </Grid.Column>
            <Grid.Column width={3} floated="right" textAlign="center">
              <SectionIcon
                className={
                  showInput ? "fa fa-chevron-circle-up" : "fa fa-plus-circle"
                }
                onClick={() => setShowInput(!showInput)}
              />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Card.Content>

      {showInput && <AddEditorForm />}
    </React.Fragment>
  );
};

export default EditorsSectionHeader;
