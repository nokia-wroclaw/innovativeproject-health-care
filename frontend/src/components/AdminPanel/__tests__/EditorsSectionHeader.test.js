import React from "react";
import { shallow } from "enzyme";
import { EditorsSectionHeader } from "../EditorsSectionHeader";

describe("EditorsSectionHeader", () => {
  it("should match snapshot", () => {
    expect(shallow(<EditorsSectionHeader />)).toMatchSnapshot();
  });
});
