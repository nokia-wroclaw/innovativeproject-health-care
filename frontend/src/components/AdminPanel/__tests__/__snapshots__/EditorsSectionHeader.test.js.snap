// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorsSectionHeader should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditorsSectionHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardContent>
          <Grid>
            <GridRow
              columns={3}
            >
              <GridColumn
                floated="left"
                width={13}
              >
                <Header
                  as="h4"
                >
                  Editors
                </Header>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="center"
                width={3}
              >
                <SectionIcon
                  className="fa fa-plus-circle"
                  onClick={[Function]}
                />
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Grid>
            <GridRow
              columns={3}
            >
              <GridColumn
                floated="left"
                width={13}
              >
                <Header
                  as="h4"
                >
                  Editors
                </Header>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="center"
                width={3}
              >
                <SectionIcon
                  className="fa fa-plus-circle"
                  onClick={[Function]}
                />
              </GridColumn>
            </GridRow>
          </Grid>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridRow
              columns={3}
            >
              <GridColumn
                floated="left"
                width={13}
              >
                <Header
                  as="h4"
                >
                  Editors
                </Header>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="center"
                width={3}
              >
                <SectionIcon
                  className="fa fa-plus-circle"
                  onClick={[Function]}
                />
              </GridColumn>
            </GridRow>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridColumn
                  floated="left"
                  width={13}
                >
                  <Header
                    as="h4"
                  >
                    Editors
                  </Header>
                </GridColumn>,
                <GridColumn
                  floated="right"
                  textAlign="center"
                  width={3}
                >
                  <SectionIcon
                    className="fa fa-plus-circle"
                    onClick={[Function]}
                  />
                </GridColumn>,
              ],
              "columns": 3,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Header
                    as="h4"
                  >
                    Editors
                  </Header>,
                  "floated": "left",
                  "width": 13,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "h4",
                    "children": "Editors",
                  },
                  "ref": null,
                  "rendered": "Editors",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <SectionIcon
                    className="fa fa-plus-circle"
                    onClick={[Function]}
                  />,
                  "floated": "right",
                  "textAlign": "center",
                  "width": 3,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "fa fa-plus-circle",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardContent>
            <Grid>
              <GridRow
                columns={3}
              >
                <GridColumn
                  floated="left"
                  width={13}
                >
                  <Header
                    as="h4"
                  >
                    Editors
                  </Header>
                </GridColumn>
                <GridColumn
                  floated="right"
                  textAlign="center"
                  width={3}
                >
                  <SectionIcon
                    className="fa fa-plus-circle"
                    onClick={[Function]}
                  />
                </GridColumn>
              </GridRow>
            </Grid>
          </CardContent>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Grid>
              <GridRow
                columns={3}
              >
                <GridColumn
                  floated="left"
                  width={13}
                >
                  <Header
                    as="h4"
                  >
                    Editors
                  </Header>
                </GridColumn>
                <GridColumn
                  floated="right"
                  textAlign="center"
                  width={3}
                >
                  <SectionIcon
                    className="fa fa-plus-circle"
                    onClick={[Function]}
                  />
                </GridColumn>
              </GridRow>
            </Grid>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GridRow
                columns={3}
              >
                <GridColumn
                  floated="left"
                  width={13}
                >
                  <Header
                    as="h4"
                  >
                    Editors
                  </Header>
                </GridColumn>
                <GridColumn
                  floated="right"
                  textAlign="center"
                  width={3}
                >
                  <SectionIcon
                    className="fa fa-plus-circle"
                    onClick={[Function]}
                  />
                </GridColumn>
              </GridRow>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <GridColumn
                    floated="left"
                    width={13}
                  >
                    <Header
                      as="h4"
                    >
                      Editors
                    </Header>
                  </GridColumn>,
                  <GridColumn
                    floated="right"
                    textAlign="center"
                    width={3}
                  >
                    <SectionIcon
                      className="fa fa-plus-circle"
                      onClick={[Function]}
                    />
                  </GridColumn>,
                ],
                "columns": 3,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Header
                      as="h4"
                    >
                      Editors
                    </Header>,
                    "floated": "left",
                    "width": 13,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "as": "h4",
                      "children": "Editors",
                    },
                    "ref": null,
                    "rendered": "Editors",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <SectionIcon
                      className="fa fa-plus-circle"
                      onClick={[Function]}
                    />,
                    "floated": "right",
                    "textAlign": "center",
                    "width": 3,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "fa fa-plus-circle",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
